component/kafka/kafka.go

```diff
@@ -0,0 +1,73 @@
+package kafka
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/Shopify/sarama"
+	"github.com/beatlabs/patron/correlation"
+	"github.com/google/uuid"
+)
+
+// FailStrategy type definition.
+type FailStrategy int
+
+const (
+	// ExitStrategy does not commit failed message offsets and exits the application.
+	ExitStrategy FailStrategy = iota
+	// SkipStrategy commits the offset of messages that failed processing, and continues processing.
+	SkipStrategy
+)
+
+// BatchProcessorFunc definition of a batch async processor.
+type BatchProcessorFunc func(context.Context, []MessageWrapper) error
+
+// MessageWrapper interface for wrapping messages that are handled by the kafka component.
+type MessageWrapper interface {
+	GetConsumerMessage() *sarama.ConsumerMessage
+	GetCorrelationID() string
```

Comment:
Every Message in SQS contains also the context, which contains the correlation id so there is no need to expose it.

Reply:
That's cool! I will do that. But what if a developer needs one single context to identify a message? like for example it is a single write to an SQL database without `n + 1` and they want to do `log.FromContext(batch.ctx)`. Or do you see that as unnecessary?